pliers/transformers/microsoft.py

```diff
@@ -0,0 +1,80 @@
+import os
+import requests
+from pliers.transformers import Transformer
+from pliers.utils import EnvironmentKeyMixin
+
+
+BASE_URL = 'https://{location}.api.cognitive.microsoft.com/{api}/{version}'\
+           '/{method}'
+
+
+class MicrosoftAPITransformer(Transformer):
+    ''' Base MicrosoftAPITransformer class.
+
+    Args:
+        subscription_key (str): a valid subscription key for Microsoft Cognitive
+            Services. Only needs to be passed the first time the extractor is
+            initialized.
+        location (str): region the subscription key has been registered in.
+        api_version (str): API version to use.
+    '''
+
+    _log_attributes = ('api_version',)
+
+    def __init__(self, subscription_key=None, location=None,
+                 api_version='v1.0'):
+        if subscription_key is None:
+            if self._env_keys not in os.environ:
+                raise ValueError("No Microsoft Cognitive Services credentials "
+                                 "found. A Microsoft Azure cognitive service "
+                                 "account key must be either passed as the "
+                                 "subscription_key argument, or set in the "
+                                 "appropriate environment variable.")
+            subscription_key = os.environ[self._env_keys]
+
+        if location is None:
+            if 'MICROSOFT_SUBSCRIPTION_LOCATION' not in os.environ:
```

Comment:
Oh, this addresses my question above. Looks like it's locked to a region. Ignore my above comment (except the part about clarifying what the string is).